File: java/src/Problem7.java
import java.util.TreeSet;

/**
 * Created by adam on 2/1/14.
 */
public class Problem7 implements Euler {
    final int max = 10001;
    TreeSet<Integer> knownPrimes;
    public Problem7() {
        knownPrimes = new TreeSet<Integer>();
        knownPrimes.add(2);
        knownPrimes.add(3);
    }
    /** Problem 7
     *
      * @return the answer
     */
    public String answer() {
        int i = knownPrimes.last()+1;
        while(true) {
            if (notInKnownPrimes(i)) {
                knownPrimes.add(i);
            }
            i++;
            if (knownPrimes.size() == max) {
                return Integer.toString(knownPrimes.last());
            }
        }
    }

    private boolean notInKnownPrimes(int test) {
        for(int x : knownPrimes) {
            if (test % x == 0) {
                return false;
            }
        }
        return true;
    }
}

File: dart/lib/src/Problem7.dart
import 'Euler.dart';
import 'EulerHelper.dart';

class Problem7 implements Euler {
  String answer() {
    int counter = 1;
    int primesSoFar = 0;
    while (true) {
      if (primesSoFar == 10001) {
        return counter.toString();
      }
      counter++;
      if (EulerHelper.is_prime(counter)) {
        primesSoFar++;
      }
    }
  }
}
